ql/time/schedule.hpp

```diff
@@ -151,7 +151,20 @@ namespace QuantLib {
         Date firstDate_, nextToLastDate_;
     };
 
+    /*! Return the CDS maturity date given the CDS trade date, \p tradeDate, the CDS \p tenor and a CDS \p rule.
 
+        A \c Null<Date>() is returned when a \p rule of \c CDS2015 and a \p tenor length of zero fail to yield a valid 
+        CDS maturity date.
+
+        \warning An exception will be thrown if the \p rule is not \c CDS2015, \c CDS or \c OldCDS.
+
+        \warning An exception will be thrown if the \p rule is \c OldCDS and a \p tenor of 0 months is provided. This 
+                 restriction can be removed if 0M tenor was available before the CDS Big Bang 2009.
+
+        \warning An exception will be thrown if the \p tenor is not a multiple of 3 months. For the avoidance of 
+                 doubt, a \p tenor of 0 months is supported.
+    */
+    Date cdsMaturity(const Date& tradeDate, const Period& tenor, DateGeneration::Rule rule);
```

Comment:
I would move this into some other file, as it's CDS-related logic and `Schedule` is more general.  Maybe with the CDS instrument?

Reply:
Yes, thanks, that makes more sense. I have done this in a74f204. The only minor issue was that I had to make the `previousTwentieth` function visible i.e. took it from the anonymous namespace in schedule.cpp and added a declaration in schedule.hpp. Hope this is ok.

Reply:
That's ok.  If you want to still hide it somewhat, you can put it into `namespace detail`.  Thanks!

Reply:
Not sure if it _should_ be hidden now, though, since it's used elsewhere.  I'd probably leave it visible.